### PR TITLE
Update bqetl query command to handle generated SQL

### PR DESCRIPTION
**Pull request by scholtzan**
For https://github.com/mozilla/bigquery-etl/issues/2313

Checklist for reviewer:

- [ ] Commits should reference a bug or github issue, if relevant (if a bug is referenced, the pull request should include the bug number in the title)
- [ ] If the PR comes from a fork, trigger integration CI tests by running the [Push to upstream workflow](https://github.com/mozilla/bigquery-etl/actions/workflows/push-to-upstream.yml) and provide the `<username>:<branch>` of the fork as parameter. The parameter will also show up
in the logs of the `manual-trigger-required-for-fork` CI task together with more detailed instructions.
- [ ] If adding a new field to a query, ensure that the schema and dependent downstream schemas have been updated

For modifications to schemas in restricted namespaces (see [`CODEOWNERS`](/CODEOWNERS)):
- [ ] Follow the [change control procedure](https://docs.google.com/document/d/1TTJi4ht7NuzX6BPG_KTr6omaZg70cEpxe9xlpfnHj9k/edit#heading=h.ttegrcfy18ck)

### CODE REVIEW

File: bigquery_etl/cli/utils.py [hunk 1]
```diff
@@ -104,6 +104,14 @@ def paths_matching_name_pattern(pattern, sql_path, project_id, files=("*.sql")):
     callback=is_valid_dir,
 )
 
+tmp_dir_option = click.option(
+    "--tmp_dir",
+    "--tmp-dir",
+    help="Path to temporary directory generated SQL is written to.",
+    type=click.Path(file_okay=False),
+    default=Path("/tmp/sql"),
```
Comment (jklukas) [inline bigquery_etl/cli/utils.py:112]: Can we consider generating the tmp directory name rather than a static path?

Comment (scholtzan) [reply bigquery_etl/cli/utils.py]: A temporary directory is now generated unless explicitly specified

File: bigquery_etl/cli/utils.py [hunk 2]
```diff
@@ -104,6 +105,25 @@ def paths_matching_name_pattern(pattern, sql_path, project_id, files=("*.sql")):
     callback=is_valid_dir,
 )
 
+
+def generate_tmp_dir(ctx, param, value):
+    """Generate temporary directory if no default directory is provided."""
+    if value is None:
+        tmp_dir = Path(tempfile.mkdtemp()) / "sql"
```
Comment (jklukas) [inline bigquery_etl/cli/utils.py:112]: Do we ever clean up this directory? If we're generating a random tmp dir, I'd expect this code to also handle cleaning up the directory when it's done.

Comment (jklukas) [reply bigquery_etl/cli/utils.py]: Can we use the pattern described in https://click.palletsprojects.com/en/8.0.x/advanced/#managing-resources ?

Comment (scholtzan) [reply bigquery_etl/cli/utils.py]: Interesting, yes! It seems like the context object cannot be renamed, e.g. I was hoping `ctx.tmp_dir` would be allowed, but only `ctx.obj` works.

Comment (scholtzan) [reply bigquery_etl/cli/utils.py]: Ah actually `ctx.obj["TMP_DIR"]` might be a good alternative here

**Review by jklukas — APPROVED** [review]
Changes look great!

**Pull request merged**